swig/cpp/src/Tree_Data.hpp

```diff
@@ -37,6 +37,12 @@ namespace libyang {
  * Class wrappers for data structures and functions to manipulate and access instance data tree.
  */
 
+struct Decimal64
+{
+    int64_t value;
+    uint8_t digits;
+};
+
```

Comment:
Not sure what the C++ convention is but I would expect a `pair` instead of this structure. @jktjkt ?

Reply:
I think that a custom `struct` is better in this context because it gives a clear meaning to both values of that tuple. If it was a `std::pair`, the usage would look like, e.g.:
```c++
auto v = val->dec64();
double res = v.first * std::pow(10, -v.second);
```

I think that the following is more readable:
```c++
auto v = val->dec64();
double res = v.value * std::pow(10, -v.digits);
```

Reply:
Alright.